bin/i18n/sync-in.rb

```diff
@@ -440,7 +441,41 @@ def localize_course_offerings
   CourseOffering.all.sort.each do |co|
     hash[co.key] = co.display_name
   end
-  File.open(File.join(I18N_SOURCE_DIR, "dashboard/course_offerings.json"), "w+") do |f|
+  write_dashboard_json('course_offerings', hash)
+end
+
+# Encapsulation of the sync-in process for our `Script`
+# model's attributes. This is meant to be built on top of
+# for any future attributes we end up needing to translate.
+def localize_scripts
+  puts "Preparing scripts"
+
+  scripts_hash = Hash.new
+  Script.all.each do |script|
+    next unless ScriptConstants.i18n? script.name
+
+    # Announcements
+    scripts_hash['script_announcements'] = Hash.new if scripts_hash['script_announcements'].nil?
+    script.announcements&.each do |announcement|
+      # An announcement is not translatable if it doesn't have a key
+      next unless announcement['key']
```

Comment:
There are ~5 cases that are _expected_. They were unable to be backfilled with a key because they are on some of the first scripts created, and now can't be updated due to the record failing validation on save. We enforce script `name` rules that those old ones no longer follow. Being that they are some of our original scripts I assumed them not having translations is okay. After this update it won't be possible to create `announcements` without a key.